### PR TITLE
refactor: extract call tx flow

### PR DESCRIPTION
**Pull request by RetricSu**
`eth_call` and `eth_estimateGas` required some typical process to normalize the tx params, this PR refactors such process to help better maintain the code.

### CODE REVIEW

File: packages/api-server/src/convert-tx.ts [hunk 1]
```diff
@@ -44,6 +46,145 @@ export interface PolyjuiceTransaction {
   s: HexString;
 }
 
+// execute raw transaction
+export async function ethCallTxToGodwokenRawTx(
```
Comment (classicalliu) [inline packages/api-server/src/convert-tx.ts:50]: Auto create account tx support removed?

File: packages/api-server/src/convert-tx.ts [hunk 2]
```diff
@@ -44,6 +46,145 @@ export interface PolyjuiceTransaction {
...
+  fromId: number,
+  toId: number,
+  nonce: number,
+  args: string
+) {
+  const rawL2Transaction = {
+    chain_id: "0x" + chainId.toString(16),
+    from_id: "0x" + BigInt(fromId).toString(16),
+    to_id: "0x" + BigInt(toId).toString(16),
+    nonce: "0x" + BigInt(nonce).toString(16),
```
Comment (classicalliu) [inline packages/api-server/src/convert-tx.ts:181]:
```suggestion
    from_id: "0x" + fromId.toString(16),
    to_id: "0x" + toId.toString(16),
    nonce: "0x" + nonce.toString(16),
```

File: packages/api-server/src/methods/normalizer.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      gas,
+      gasPrice,
+    };
+  }
+
+  async normalizeEstimateGasTx(
+    txEstimateGasObj: Partial<TransactionCallObject>
+  ): Promise<Required<TransactionCallObject>> {
+    const to = txEstimateGasObj.to || "0x";
+    return this.normalizeCallTx({ ...{ to }, ...txEstimateGasObj });
```
Comment (classicalliu) [inline packages/api-server/src/methods/normalizer.ts:80]:
```suggestion
    return this.normalizeCallTx({ to, ...txEstimateGasObj });
```

File: packages/api-server/src/convert-tx.ts (hunk 2 above)
Comment (RetricSu) [reply packages/api-server/src/convert-tx.ts]: updated

File: packages/api-server/src/methods/normalizer.ts (hunk 3 above)
Comment (RetricSu) [reply packages/api-server/src/methods/normalizer.ts]: updated

**Review by classicalliu — APPROVED** [review]

File: packages/api-server/src/methods/normalizer.ts [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    gwConfig.accounts.defaultFrom.scriptHash
+  );
+  if (defaultFromScript == null) {
+    throw new Error("default from script is null");
+  }
+  const defaultFromAddress = "0x" + defaultFromScript.args.slice(2).slice(64);
+  return defaultFromAddress;
+}
+
+export async function getMaxGasByBalance(
```
Comment (keroro520) [inline packages/api-server/src/methods/normalizer.ts:95]: unused function?

**Review by keroro520 — APPROVED** [review]

File: packages/api-server/src/methods/normalizer.ts (hunk 4 above)
Comment (RetricSu) [reply packages/api-server/src/methods/normalizer.ts:95]: the follow up pr https://github.com/nervosnetwork/godwoken-web3/pull/520 will use it

**Pull request merged**